pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/hashing/XXH3_64.java

```diff
@@ -0,0 +1,443 @@
+// SPDX-License-Identifier: Apache-2.0
+package com.hedera.pbj.runtime.hashing;
+
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.VarHandle;
+import java.nio.ByteOrder;
+
+/**
+ * XXH3_64 is a 64-bit variant of the XXH3 hash function.
+ * It is designed to be fast and efficient, providing a good balance between speed and
+ * collision resistance.
+ *
+ * <p>It is recommended to use {@link #DEFAULT_INSTANCE} for most use cases.
+ * @see <a href="https://xxhash.com">xxhash</a>
+ */
+@SuppressWarnings({"DuplicatedCode", "NumericOverflow"})
+public class XXH3_64 {
+    /** Default instance of the XXH3_64 hasher with a seed of 0. */
+    public static final XXH3_64 DEFAULT_INSTANCE = new XXH3_64(0);
+
+    private static final VarHandle LONG_HANDLE =
```

Comment:
We already have some methods to get longs from byte arrays using `Unsafe`, see `UnsafeUtils` class for details. I wonder if these methods can be used here. The idea was to have everything unsafe in a single place...